orchagent/portsorch.cpp

```diff
@@ -4021,6 +4021,50 @@ void PortsOrch::initializeQueues(Port &port)
     SWSS_LOG_INFO("Get queues for port %s", port.m_alias.c_str());
 }
 
+void PortsOrch::initializeSchedulerGroups(Port &port)
+{
+    std::vector<sai_object_id_t> scheduler_group_ids;
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_QOS_NUMBER_OF_SCHEDULER_GROUPS;
+    sai_status_t status = sai_port_api->get_port_attribute(port.m_port_id, 1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to get number of scheduler groups for port:%s", port.m_alias.c_str());
+        task_process_status handle_status = handleSaiGetStatus(SAI_API_PORT, status);
+        if (handle_status != task_process_status::task_success)
+        {
+            throw runtime_error("PortsOrch initialization failure.");
+        }
+    }
+    SWSS_LOG_INFO("Got %d number of scheduler groups for port %s", attr.value.u32, port.m_alias.c_str());
+
+    scheduler_group_ids.resize(attr.value.u32);
+
+    if (attr.value.u32 == 0)
+    {
+        return;
+    }
+
+    attr.id = SAI_PORT_ATTR_QOS_SCHEDULER_GROUP_LIST;
+    attr.value.objlist.count = (uint32_t)scheduler_group_ids.size();
+    attr.value.objlist.list = scheduler_group_ids.data();
```

Comment:
why not storing `scheduler_group_ids` into the port? We need to fetch it when handle `QUEUE` table in qosorch.

Reply:
Yes, this can be stored into port similar to queue ids and used by qosorch. Can this be taken up as separate PR. Thanks

Reply:
There is a logic in [getSchedulerGroup](https://github.com/Azure/sonic-swss/blob/master/orchagent/qosorch.cpp#L1278) which is exactly the same as this one.
It will take more time for the system to start if scheduler group IDs are fetched for all ports twice, which can introduce degradation in fast reboot.

Reply:
As discussed offline, the comment will be addressed in another PR

Reply:
@arvbb , based on Stephen's comment above on the potential impact to the fast reboot time, has this change been tested against fast reboot?

Reply:
Hi Neetha,  I did not notice any impact in my testing with the default system configuration. Also CI seems to be fine.